hybrasyl/Messaging/Admin.cs

```diff
@@ -0,0 +1,410 @@
+    }
+
+    class DebugCommand : ChatCommand
+    {
+        public new static string Command = "debug";
+        public new static string ArgumentText = "None";
+        public new static string HelpText = "Toggle whether or not debugging is enabled on the server.";
+        public new static bool Privileged = true;
+
+        public new static ChatCommandResult Run(User user, params string[] args)
+        {
+            var enabled = Game.World.ToggleDebug();
+            if (enabled)
+                return Success("Debugging enabled");
+            return Success("Debugging disabled");
+        }
+    }
+
+    class SpawnCommand : ChatCommand
+    {
+        public new static string Command = "spawn";
+        public new static string ArgumentText = "<string monsterName> [<uint hp> <uint mp> <uint str> <uint int> <uint wis> <uint con> <uint dex> <uint xp> <uint gold>]";
+        public new static string HelpText = "Spawn the specified monster at your current coordinates, with optionally specified stats.";
+        public new static bool Privileged = true;
+
+        public new static ChatCommandResult Run(User user, params string[] args)
+        {
+
+            if (Game.World.WorldData.TryGetValue(args[0], out Creatures.Creature creature))
+            {
+                Spawn spawn = new Spawn();
+                spawn.Stats.Hp = 100;
+                spawn.Stats.Mp = 100;
+                spawn.Stats.Str = 3;
+                spawn.Stats.Int = 3;
+                spawn.Stats.Wis = 3;
+                spawn.Stats.Con = 3;
+                spawn.Stats.Dex = 3;
+                spawn.Loot.Xp.Min = 1;
+                spawn.Loot.Xp.Max = 1;
+                spawn.Loot.Gold.Min = 1;
+                spawn.Loot.Gold.Min = 1;
+                if (args.Length >= 2)
+                    spawn.Stats.Hp = uint.Parse(args[1]);
+                if (args.Length >= 3)
+                    spawn.Stats.Mp = uint.Parse(args[2]);
+                if (args.Length >= 4)
+                    spawn.Stats.Str = byte.Parse(args[3]);
+                if (args.Length >= 5)
+                    spawn.Stats.Int = byte.Parse(args[4]);
+                if (args.Length >= 6)
+                    spawn.Stats.Wis = byte.Parse(args[5]);
+                if (args.Length >= 7)
+                    spawn.Stats.Con = byte.Parse(args[6]);
+                if (args.Length >= 8)
+                    spawn.Stats.Dex = byte.Parse(args[7]);
+                if (args.Length >= 9)
+                {
+                    spawn.Loot.Xp.Min = byte.Parse(args[8]);
+                    spawn.Loot.Xp.Max = byte.Parse(args[8]);
+                }
+                if (args.Length >= 10)
+                {
+                    spawn.Loot.Gold.Min = byte.Parse(args[9]);
+                    spawn.Loot.Gold.Max = byte.Parse(args[9]);
+                }
+                Monster newMob = new Monster(creature, spawn, user.MapId);
+                user.Map.Insert(newMob, user.X, user.Y);
```

Comment:
Does the newMob have be inserted in the world data as well as the Map? I seem to be unable to attack or target (spell cast) the newMob.

Reply:
fixed